src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -488,11 +488,34 @@ public boolean openPairingWindowWithPINCallback(
         deviceControllerPtr, devicePtr, duration, iteration, discriminator, setupPinCode, callback);
   }
 
-  /* Shutdown all cluster attribute subscriptions for a given device */
-  public void shutdownSubscriptions(long devicePtr) {
-    shutdownSubscriptions(deviceControllerPtr, devicePtr);
+  public int getFabricIndex() {
+    return getFabricIndex(deviceControllerPtr);
   }
 
+  /* Shutdown all subscriptions */
+  /* if schedule is true, it would be forced to run in matter thread */
+  public void shutdownSubscriptions(boolean schedule) {
+    shutdownSubscriptions(deviceControllerPtr, null, null, null, schedule);
+  }
+
+  /* Shutdown the subscriptions based upon fabricIndex,  */
+  /* if schedule is true, it would be forced to run in matter thread */
+  public void shutdownSubscriptions(Integer fabricIndex, boolean schedule) {
```

Comment:
Since we've split up the APIs these parameters could be primitives rather than wrappers.

Reply:
rollback the version without schedule and update the comments. thanks